### PR TITLE
add applicability checks to fusion solver

### PR DESCRIPTION
**Pull request by JehandadKhan**
This PR resolves a problem with #1889

### CODE REVIEW

**Comment by junliume** [comment]
@bghimireamd could you add a test to this fix? It could be in a separate PR due to urgency.

File: test/gtest/bad_fusion_plan.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+{
+
+public:
+    TestFusionPlan(const miopenTensorLayout_t& tensor_layout,
+                   const miopenActivationMode_t& activ_mode)
+        : handle(get_handle())
+    {
+
+        input   = tensor<T>{miopen_type<T>{}, tensor_layout, conv_config.GetInput()};
+        weights = tensor<T>{miopen_type<T>{}, tensor_layout, conv_config.GetWeights()};
```
Comment (JehandadKhan) [inline test/gtest/bad_fusion_plan.cpp]: If we are just checking applicability then we dont need the full tensor, just the descriptor

File: test/gtest/bad_fusion_plan.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+        std::random_device rd{};
+        std::mt19937 gen{rd()};
+        std::uniform_real_distribution<> d{-3, 3};
+        auto gen_value = [&](auto...) { return d(gen); };
+        input.generate(gen_value);
+        weights.generate(gen_value);
```
Comment (JehandadKhan) [inline test/gtest/bad_fusion_plan.cpp]: this would be unnecessary as well

File: test/gtest/bad_fusion_plan.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        activ_desc = {activ_mode, activ_alpha, activ_beta, activ_gamma};
+        conv_desc  = conv_config.GetConv();
+        miopen::TensorDescriptor output_desc =
+            conv_desc.GetForwardOutputTensor(input.desc, weights.desc, GetDataType<T>());
+        output = tensor<T>{miopen_type<T>{}, tensor_layout, output_desc.GetLengths()};
+        bias   = tensor<T>{1, static_cast<size_t>(conv_config.k), 1, 1};
+        bias.generate(gen_value);
+        std::fill(output.begin(), output.end(), std::numeric_limits<double>::quiet_NaN());
```
Comment (JehandadKhan) [inline test/gtest/bad_fusion_plan.cpp]: And this

**Comment by JehandadKhan** [comment]
@bghimireamd  Thanks for the quick turnaround, it looks good to me. I cant approve however, since I am the author of the PR.

cc @junliume

**Review by junliume — DISMISSED** [review]

File: src/solver/conv_ck_igemm_fwd_bias_activ_fused.cpp [hunk 4]
```diff
@@ -387,9 +387,26 @@ bool ConvCKIgemmFwdBiasActivFused::IsApplicable(const FusionContext& ctx,
     std::ignore = fdesc_problem;
     return false;
 #else
-    const auto& problem = fdesc_problem.GetConvProblem(0, conv::Direction::Forward);
+    const auto& desc = *fdesc_problem.fusion_plan_desc;
+    if(desc.op_map.empty())
+    {
+        MIOPEN_THROW("");
```
Comment (atamazov) [inline src/solver/conv_ck_igemm_fwd_bias_activ_fused.cpp]: [R]
```suggestion
        MIOPEN_THROW(miopenStatusInternalError, "desc.op_map.empty()");
```

**Review by junliume — APPROVED** [review]

**Pull request merged**